drafts/build_a_dao_on_fuel/draft.md

```diff
@@ -0,0 +1,1459 @@
+
+1. [Installation](https://docs.fuel.network/guides/installation/) of dependencies like `fuel`, `sway` extensions on vscode, `rustc`, `cargo`, `rustup` `cargo-generate`, on your pc.
+2. Node version ^18.18.2 || ^20.0.0.
+3. You should have `wsl` installed if you are using a Windows or Mac PC; you can use the code `wsl --install` to download `wsl`.
+4. Knowledge of fuelVM and other fuel tools.
+5. Basic concept of blockchain technology.
+6. Writing, compiling, and deploying to any blockchain (EVM or non-EVM compatible blockchain).
+7. Understand concepts of ABIs, Faucets, wallet connect, etc.
+
+## Getting Started
+
+Before you delve into the juicy part, here are some concepts to jug your memory on the fuel ecosystem and help you better understand words you will see throughout this tutorial.
+
+**Fuel**: [Fuel](https://fuel.network/) is an operating system purpose-built for Ethereum rollups, designed to help developers build decentralized economies at scale.
+
+**FuelVM**: The [Fuel Virtual Machine](https://medium.com/blockchain-capital-blog/exploring-the-fuelvm-86cf9ccdc159) responsible for managing all code execution on the Fuel Blockchain
+
+**Sway**: [Sway](https://fuellabs.github.io/sway/) is a domain-specific language used for writing smart contracts on the fuel blockchain.
+
+**ABI**: These are called [Application Binary Interfaces](https://docs.fuel.network/guides/intro-to-sway/contract-abi/); they are the blueprint or interaction models of your smart contract, it is a layout of the detailed functionalities in your smart contract, and how to interact with it.
+
+## Setting up your Environment
+
+In the next few steps, you will set up your development environment to begin building your smart contract and the front end of your Dao.
+
+1. Go to your terminal and run the command `wsl` to start your `wsl` terminal
+
+2. Next, run the command to generate your new contract directory.
+
+```bash
+mkdir Fuel_Dao
+cd Fuel_Dao
+forc new contract
+```
+
+## Writing your Smart Contract
+
+Head to the `main.sw` contract file, where you will implement your main contracts. Following the next few steps, you will replace the code therein.
+
+### 1. Importing from the modules 
+
+First, you will import the following modules from the sway standard library; 
+  
+- `asset::transfer`: This provides the functionality for transferring assets.
+
+- `auth::msg_sender` - The module retrieves the address of the sender of the contract or a transaction.
+  
+- `block::height` - The module provides information about the current block height or block number
+  
+- `call_frames::msg_asset_id`- This provides information about the asset (token) associated with the current sent transaction.
+  
+- `context` - This module provides context-related functionality related to the execution context of your contract.
+  
+`hash::Hash` - This provides hashing-related functionality, such as integrity verification, cryptographic operations, and generating unique identifiers.
```

Comment:
Missing hyphen at the beggining of the phrase.

Reply:
Thanks for spotting this out 
- Corrected 
> -  `hash::Hash`: This provides hashing-related functionality, such as integrity verification, cryptographic operations, and generating unique identifiers.